packages/@aws-cdk/aws-ecr/lib/lifecycle.ts

```diff
@@ -0,0 +1,135 @@
+/**
+ * An ECR life cycle rule
+ */
+export interface LifecycleRule {
+    /**
+     * Controls the order in which rules are evaluated (low to high)
+     *
+     *  When you add rules to a lifecycle policy, you must give them each a
+     *  unique value for rulePriority. Values do not need to be sequential
+     *  across rules in a policy. A rule with a tagStatus value of any must have
+     *  the highest value for rulePriority and be evaluated last.
+     */
+    rulePriority: number;
+
+    /**
+     * Describes the purpose of the rule
+     *
+     * @default No description
+     */
+    description?: string;
+
+    /**
+     * Select images based on tags
+     *
+     * Only one rule is allowed to select untagged images, and it must
+     * have the highest rulePriority.
+     *
+     * @default TagStatus.Any
+     */
+    tagStatus?: TagStatus;
+
+    /**
+     * Select images that have ALL the given prefixes in their tag.
+     *
+     * Only if tagStatus == TagStatus.Any
+     */
+    tagPrefixList?: string[];
+
+    /**
+     * Specify limit type to apply to the repository
+     */
+    countType: CountType;
+
+    /**
+     * The unit of specifying the count
+     *
+     * Only applies when countType = CountType.SinceImagePushed.
+     *
+     * @default CountUnit.Days
+     */
+    countUnit?: CountUnit;
+
+    /**
+     * The number of images or days, according to countType
+     */
+    countNumber: number;
+
+    /**
+     * The action to take
+     *
+     * @default Action.expire
+     */
+    action?: Action;
+}
+
+/**
+ * Select images based on tags
+ */
+export enum TagStatus {
+    /**
+     * Rule applies to all images
+     */
+    Any = 'any',
+
+    /**
+     * Rule applies to tagged images
+     */
+    Tagged = 'tagged',
+
+    /**
+     * Rule applies to untagged images
+     */
+    Untagged = 'untagged',
+}
+
+/**
+ * Select images based on counts
+ */
+export enum CountType {
+    /**
+     * Set a limit on the number of images in your repository
+     */
+    ImageCountMoreThan = 'imageCountMoreThan',
+
+    /**
+     * Set an age limit on the images in your repository
+     */
+    SinceImagePushed = 'sinceImagePushed',
+}
+
+/**
+ * The unit to count time in
+ */
+export enum CountUnit {
```

Comment:
Why do we even need this enum if it only has a single member?

Reply:
Yeah, I was on the fence about that as well myself. My thought was we'd automatically support new enum members as soon as they get added upstream (even though we might lack syntactical support).

But I guess that doesn't hold over JSII anyway.